tools/permission_prompt_test.py

```diff
@@ -5,220 +5,185 @@
+        assert b'⚠️  Deno requests' in stderr
+        assert b'First check failed' in stdout
+        assert not b'PermissionDenied: permission denied' in stderr
 
-    def test_write_no_prompt(self):
-        code, _stdout, stderr = self.run('needsWrite', b'', no_prompt=True)
-        assert code == 1
-        assert b'PermissionDenied: permission denied' in stderr
-
-    def test_env_yes(self):
-        code, stdout, stderr = self.run('needsEnv', b'y\n')
-        assert code == 0
-        assert stdout == b''
-        assert b'⚠️  Deno requests access to environment' in stderr
-
-    def test_env_arg(self):
-        code, stdout, stderr = self.run('needsEnv', b'', allow_env=True)
+    def test_allow(self, test_type):
+        code, stdout, stderr = self.run("needs" + test_type.capitalize(),
+                                        b'a\n')
         assert code == 0
-        assert stdout == b''
-        assert stderr == b''
-
-    def test_env_no(self):
-        code, _stdout, stderr = self.run('needsEnv', b'N\n')
-        assert code == 1
-        assert b'PermissionDenied: permission denied' in stderr
-        assert b'⚠️  Deno requests access to environment' in stderr
+        assert b'⚠️  Deno requests' in stderr
+        assert not b'First check failed' in stdout
+        assert not b'PermissionDenied: permission denied' in stderr
 
-    def test_env_no_prompt(self):
-        code, _stdout, stderr = self.run('needsEnv', b'', no_prompt=True)
+    def test_deny(self, test_type):
+        code, stdout, stderr = self.run("needs" + test_type.capitalize(),
+                                        b'd\n')
         assert code == 1
+        assert b'⚠️  Deno requests' in stderr
+        assert b'First check failed' in stdout
         assert b'PermissionDenied: permission denied' in stderr
 
-    def test_net_yes(self):
-        code, stdout, stderr = self.run('needsEnv', b'y\n')
+    def test_unrecognized_option(self, test_type):
+        code, stdout, stderr = self.run("needs" + test_type.capitalize(),
+                                        b'e\na\n')
         assert code == 0
-        assert stdout == b''
-        assert b'⚠️  Deno requests access to environment' in stderr
-
-    def test_net_arg(self):
-        code, stdout, stderr = self.run('needsNet', b'', allow_net=True)
-        assert code == 0
-        assert stdout == b''
-        assert stderr == b''
-
-    def test_net_no(self):
-        code, _stdout, stderr = self.run('needsNet', b'N\n')
+        assert b'⚠️  Deno requests' in stderr
+        assert not b'First check failed' in stdout
+        assert not b'PermissionDenied: permission denied' in stderr
+        assert b'Unrecognized option' in stderr
+
+    def test_no_prompt(self, test_type):
+        code, stdout, stderr = self.run("needs" + test_type.capitalize(), b'',
+                                        ["--no-prompt"])
         assert code == 1
+        assert not b'⚠️  Deno requests' in stderr
```

Comment:
This is a fairly weak assertion. 
```
assert not b'⚠️' in stderr
```
Would be stronger. It's a little disconcerting that these strings are repeated throughout this file. If we end up changing the prompt text, there are many call sites to update.
Could you set a constant at the top of the file for each of these strings, and use that constant for both positive and negative assertions?